src/runtime/account.ts

```diff
@@ -45,23 +70,97 @@ export class Account {
   }
 
   get accountId(): string {
-    return this.najAccount.accountId;
+    return this._accountId;
+  }
+
+  get prefix(): string {
+    return this.levelUp ? this.accountId.replace(`.${this.levelUp}`, "") : this.accountId;
   }
 
   async balance(): Promise<AccountBalance> {
     return this.najAccount.getAccountBalance() as Promise<AccountBalance>;
   }
 
+  createTransaction(receiver: Account | string): Transaction {
+    return new RuntimeTransaction(this.runtime, this, receiver);
+  }
+
   get provider(): nearAPI.providers.JsonRpcProvider {
     return this.connection.provider as nearAPI.providers.JsonRpcProvider;
   }
 
   async getKey(accountId: string): Promise<KeyPair> {
-    return this.keyStore.getKey(this.networkId, accountId);
+    return this.keyStore.getKey(this.networkId, this.makeSubAccount(accountId));
   }
 
   async setKey(accountId: string, keyPair: KeyPair): Promise<void> {
-    await this.keyStore.setKey(this.networkId, accountId, keyPair);
+    await this.keyStore.setKey(this.networkId, this.makeSubAccount(accountId), keyPair);
+  }
+
+  async addKey(accountId: string, keyPair?: KeyPair): Promise<PublicKey> {
+    const id = this.makeSubAccount(accountId);
```

Comment:
Since this is only used by `internalCreateAccount`, should it also be marked as `private` or `protected`?

Reply:
Hmm I'm not sure.  It might confuse people since it will remove the old key.  So let's do that for now.